app/services/target_service.rb

```diff
@@ -0,0 +1,16 @@
+class TargetService
```

Comment:
wdyt about renaming to `CreateTargetService`?
or if you want to have one service with all actions, then renaming the `call` action to `create`. My point is that `TargetService.new(...).call` isn't specifying what are you doing with the target, if creating, upserting, removing, etc.

Reply:
I understand your point and I agree with you. In this case, I prefer the second implementation because if in the future I want to include other actions in the target´s resource, I have all the logic in one place that has all the necessary information.